core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -15895,8 +15895,8 @@ private boolean hasIdentityClaimInitially(List<ExpressionCondition> expressionCo
 
             for (ExpressionCondition expressionCondition : expressionConditions) {
                 List<org.wso2.carbon.user.api.ClaimMapping> mappedClaim =
-                        Arrays.stream(claimMapping).filter(mapping -> mapping.getMappedAttribute() ==
-                                expressionCondition.getAttributeName()).collect(Collectors.toList());
+                        Arrays.stream(claimMapping).filter(mapping -> mapping.getMappedAttribute()
+                                .equals(expressionCondition.getAttributeName())).collect(Collectors.toList());
```

Comment:
Can the mapping.getMappedAttribute() can be null. if so, shall we add null check mapping.getMappedAttribute().